tools/src/bin/generate-modpow-tests.rs

```diff
@@ -0,0 +1,85 @@
+use std::fs;
+
+use clvmr::Allocator;
+use num_bigint::BigInt;
+use num_integer::Integer;
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+fn main() {
+    // Seed the RNG with a fixed value for reproducibility.
+    let mut rng = ChaCha8Rng::seed_from_u64(1337);
+
+    // Generate a random quantity (within a range) of random bytes.
+    let mut bytes = |min: usize, max: usize| {
+        let len = rng.gen_range(min..=max);
+        let mut bytes = vec![0; len];
+        rng.fill(&mut bytes[..]);
+        bytes
+    };
+
+    let mut tests =
+        "; This file was generated by tools/src/bin/generate-modpow-tests.rs\n\n".to_string();
+
+    for _ in 0..100 {
+        let base = BigInt::from_signed_bytes_be(&bytes(0, 32));
+
+        // Generate a random exponent, but ensure it's positive.
+        let mut exponent = BigInt::from_signed_bytes_be(&bytes(0, 32));
+        if exponent < BigInt::ZERO {
+            exponent = -exponent;
+        }
```

Comment:
isn't there a way to treat the bytes as unsigned to begin with?

Reply:
Changed it to `from_bytes_be(Sign::Plus, ...)`